src/test/ui/unsized/issue-91801.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0746]: return type cannot have an unboxed trait object
+  --> $DIR/issue-91801.rs:8:77
+   |
+LL | fn or<'a>(first: &'static Validator<'a>, second: &'static Validator<'a>) -> Validator<'a> {
+   |                                                                             ^^^^^^^^^^^^^ doesn't have a size known at compile-time
+   |
+   = note: for information on `impl Trait`, see <https://doc.rust-lang.org/book/ch10-02-traits.html#returning-types-that-implement-traits>
+help: use `impl (dyn Fn(&'a Something) -> Result<(), ()> + Send + Sync + 'a)` as the return type, as all return paths are of type `Box<[closure@$DIR/issue-91801.rs:10:21: 12:6]>`, which implements `(dyn Fn(&'a Something) -> Result<(), ()> + Send + Sync + 'a)`
+   |
+LL | fn or<'a>(first: &'static Validator<'a>, second: &'static Validator<'a>) -> impl (dyn Fn(&'a Something) -> Result<(), ()> + Send + Sync + 'a) {
```

Comment:
This `impl (dyn Trait)` type looks fishy, doesn't it?

Reply:
whoops, forgot to strip some parenthesis. yes, it is incorrect :sweat_smile: